libsrc/_DEVELOPMENT/math/float/math32/z80/f32_fsldexp.asm

```diff
@@ -61,7 +61,7 @@ PUBLIC _m32_ldexpf
     add d
     ld d,a                      ; exponent returned
 
-    rl e                        ; get sign back
+    sla e                       ; restore sign to C
```

Comment:
I think we still need to check the exponent, and if zero, jp to m32_zero_fastcall

Reply:
yes. that is a worthwhile check. not the root cause here. but good to add :+1:

Reply:
Put it after returning the sign the the right bit. That way fszero picks up the correct sign.

Reply:
Isn't that checking that the (pw2 + exp) is zero? - i.e. the new value of the exponent. I think we need to check that incoming exponent is 0.

Reply:
Both, I'd say. Neither are permitted to succeed.

Reply:
Good point!

I’ll steal this pattern for the other libraries.